server/py/services/api/migrations/versions/6e8e4df16a4e_remove_legacy_artifacts_table.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Remove legacy Artifacts table
+
+Revision ID: 6e8e4df16a4e
+Revises: a6f358254cbd
+Create Date: 2025-07-21 09:33:01.444888
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import mysql
+
+# revision identifiers, used by Alembic.
+revision = "6e8e4df16a4e"
+down_revision = "a6f358254cbd"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_index("_artifacts_uc", table_name="artifacts")
+    op.drop_table("artifacts")
+    op.drop_index("_artifacts_labels_uc", table_name="artifacts_labels")
+    op.drop_index("idx_artifacts_labels_name_value", table_name="artifacts_labels")
+    op.drop_table("artifacts_labels")
+    op.drop_index("_artifacts_tags_uc", table_name="artifacts_tags")
+    op.drop_table("artifacts_tags")
+    # ### end Alembic commands ###
+
+
+def downgrade():
```

Comment:
could be helpful to split logic into smaller helper methods per table to improve readability, not critical at all but could make it easier to follow.